### PR TITLE
Create New Timetable When Creating Custom Event

### PR DESCRIPTION
**Pull request by JiaqiWang18**
## Description
Add code to create a new timetable on creating custom events when the user does not have any timetable

### CODE REVIEW

**Review by JJamesWWang — CHANGES_REQUESTED** [review]
For some reason, isFetching in timetablesReducer is not being set to False, so the circle in the top right is infinitely spinning. Looking into this right now.

**Comment by JJamesWWang** [comment]
Also, I just realized that we were planning to be remove personal events from the body of the saving timetable request, so might be better to find a different solution here

**Comment by JJamesWWang** [comment]
I think it might be better to leave that to a different PR though, so unless you say otherwise, let's just try to fix the fetching issue.

**Comment by JiaqiWang18** [comment]
Sure, I will look onto a different solution tomorrow

**Comment by JiaqiWang18** [comment]
> I think it might be better to leave that to a different PR though, so unless you say otherwise, let's just try to fix the fetching issue.

This is fine as well

**Comment by JJamesWWang** [comment]
There's a deeper issue with the codebase here imo. I wonder if we can combine the two timetable endpoints (/timetables) and (/user/timetables) into one. Or at least separate them so that only one is called at a time because this `isFetching` is only spinning because there's no timetables returned (as the only thing they added is a custom event). To be honest, `fetchTimetables` probably should never have been called to begin with.

**Comment by JJamesWWang** [comment]
A temporary fix I can think of is to add a new reducer that sets `isFetching` to False and extend the if chain in `timetable_actions.jsx`, line 112 with an else. I'll let you also look into it to see what you come up with.

**Comment by JiaqiWang18** [comment]
> A temporary fix I can think of is to add a new reducer that sets `isFetching` to False and extend the if chain in `timetable_actions.jsx`, line 112 with an else. I'll let you also look into it to see what you come up with.

Adding an action to set it to false seems to be the best fix without tackling the deeper issue

**Comment by JJamesWWang** [comment]
I think we can ahead with that implementation then

**Review by JJamesWWang — COMMENTED** [review]
We can probably replace the fetchingCourses reducer that only sets isFetching to true as well
Line 56 in timetablesSlice.ts

File: static/js/redux/actions/timetable_actions.jsx [hunk 1]
```diff
@@ -67,7 +67,7 @@ export const fetchTimetables =
     const state = getState();
 
     // mark that we are now asynchronously requesting timetables
```
Comment (JJamesWWang) [inline static/js/redux/actions/timetable_actions.jsx:69]: Looks good

**Review by JJamesWWang — APPROVED** [review]
Looks good

**Pull request merged**